ignite/metrics/accuracy.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import division
+
+import torch
+
+from ignite.metrics.metric import Metric
+from ignite.exceptions import NotComputableError
+
+
+class Accuracy(Metric):
+    """
+    Calculates the accuracy.
+
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y_pred` must be in the following shape (batch_size, num_categories, ...) or (batch_size, ...)
+    - `y` must be in the following shape (batch_size, ...)
+    """
+    def reset(self):
+        self._num_correct = 0
+        self._num_examples = 0
+
+    def update(self, output):
+        y_pred, y = output
+
+        assert y.ndimension() >= 1 and y_pred.ndimension() >= 1
```

Comment:
@anmolsjoshi please use the pattern `if ... then raise"

Reply:
@vfdev-5  Done! Do you want this done in tests as well?